src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs

```diff
@@ -2366,5 +2369,45 @@ void M([|D|]&lt;int&gt; lambda)
             var metadataAsSourceFile = await context.GenerateSourceAsync(navigationSymbol);
             context.VerifyResult(metadataAsSourceFile, expected);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestFailedDecompilation()
+        {
+            var exportProvider = ExportProviderCache.GetOrCreateExportProviderFactory(TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithPart(typeof(ThrowingDecompiledSourceService))).CreateExportProvider();
+            var throwingService = exportProvider.GetExportedValue<ThrowingDecompiledSourceService>();
+            var expectedException = throwingService.Exception;
+            var lazyFailureText = new LazyToString(() => "// " + expectedException.ToString().Replace("\n", "\n// "));
```

Comment:
This seems *really* fragile as a way to make this work. Can we just make the Expected take a  `Lazy<string>` so it's obvious this is intended to be deferred?

Reply:
Changed to `Lazy<string>`